apps/balancer-tools/src/app/apr/round/(components)/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,172 @@
+"use client";
+
+import { Cross1Icon, MagnifyingGlassIcon } from "@radix-ui/react-icons";
+import Downshift, { ControllerStateAndHelpers } from "downshift";
+import React, { useState } from "react";
+
+export const MultiSelectDropdown = ({
+  items,
+  labelText,
+  onSelectionItemsChange,
+  initialSelectedItems,
+  ...rest
+}: {
+  items: string[];
+  labelText: string;
+  onSelectionItemsChange: (items: string[]) => void;
+  initialSelectedItems: string[];
+}) => {
+  const [selectedItems, setSelectedItems] =
+    useState<string[]>(initialSelectedItems);
+
+  function handleOKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    const inputElement = event.target as HTMLInputElement;
+    if (!inputElement.value.length) {
+      if (event.key === "Backspace") {
+        const copySelectedItems = [...selectedItems];
+        copySelectedItems.pop();
+        setSelectedItems(copySelectedItems);
+        onSelectionItemsChange(copySelectedItems);
+      }
+    }
+  }
+
+  return (
+    <div className="relative">
+      <Downshift
+        {...rest}
+        onChange={changeHandler(
+          selectedItems,
+          setSelectedItems,
+          onSelectionItemsChange,
+        )}
+      >
+        {({
+          getLabelProps,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          toggleMenu,
+          inputValue,
+        }) => {
+          return (
+            <div>
+              <div className="relative mt-1">
+                <div className="flex items-center bg-blue6 relative w-full cursor-default overflow-hidden rounded-lg text-left shadow-md focus:outline-none focus-within:ring-2 focus-within:ring-blue6 focus-within:ring-opacity/75 focus-within:ring-offset-2 focus-within:ring-offset-blue3 sm:text-sm">
+                  <label {...getLabelProps()} className="flex-1 mx-1">
+                    <MagnifyingGlassIcon />
+                  </label>
+                  {selectedItems.map((value, idx) => {
+                    return (
+                      <button
+                        key={value + idx}
+                        className="relative mx-1 flex gap-2 h-fit items-center rounded-md px-2 py-1 bg-blue7 whitespace-nowrap"
+                        onClick={() =>
+                          removeSelectedItemByIndex(
+                            idx,
+                            selectedItems,
+                            setSelectedItems,
+                            onSelectionItemsChange,
+                          )
+                        }
+                      >
+                        <span>{value}</span>
+                        <span>
+                          <Cross1Icon />
+                        </span>
+                      </button>
+                    );
+                  })}
+
+                  <input
+                    placeholder={labelText}
```

Comment:
this is a placeholder text, not [label ](https://www.w3schools.com/tags/tag_label.asp) right